packages/rum-core/src/domain/startUserDataTelemetry.ts

```diff
@@ -0,0 +1,119 @@
+import type { BatchFlushEvent, Context, ContextManager, Observable, Telemetry } from '@datadog/browser-core'
+import {
+  isExperimentalFeatureEnabled,
+  performDraw,
+  ONE_SECOND,
+  addTelemetryDebug,
+  monitor,
+} from '@datadog/browser-core'
+import { RumEventType } from '../rawRumEvent.types'
+import type { RumEvent } from '../rumEvent.types'
+import type { RumConfiguration } from './configuration'
+import type { FeatureFlagContexts } from './contexts/featureFlagContext'
+import type { LifeCycle } from './lifeCycle'
+import { LifeCycleEventType } from './lifeCycle'
+
+export const MEASURES_FLUSH_INTERVAL = 10 * ONE_SECOND
+
+type Measure = {
+  min: number
+  max: number
+  sum: number
+}
+
+let batchCount: number
+let batchBytesCount: Measure
+let batchMessagesCount: Measure
+let globalContextBytes: Measure
+let userContextBytes: Measure
+let featureFlagBytes: Measure
+
+let currentBatchGlobalContextBytes: Measure
+let currentBatchUserContextBytes: Measure
+let currentBatchFeatureFlagBytes: Measure
+
+export function startUserDataTelemetry(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about using `CustomerData` naming? it would avoid confusion with user context and be consistent with naming used in RFC, jira, ...